disco/cli/select_time_points.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+"""
+Down-selects load shape time points in the circuit based on
+user-specified critical conditions.
+"""
+
+import logging
+import shutil
+import sys
+from pathlib import Path
+
+import click
+
+from jade.loggers import setup_logging
+from jade.utils.utils import get_cli_string
+
+from disco.preprocess.select_timepoints2 import (
+    CriticalCondition,
+    DemandCategory,
+    GenerationCategory,
+    main,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument("master_file", type=click.Path(exists=True), callback=lambda *x: Path(x[2]))
+@click.option(
+    "-c",
+    "--critical-conditions",
+    type=click.Choice([x.value for x in CriticalCondition]),
+    default=tuple(x.value for x in CriticalCondition),
+    show_default=True,
+    multiple=True,
+    callback=lambda *x: tuple(CriticalCondition(y) for y in x[2]),
+    help="critical conditions to use for time-point selection",
+)
+@click.option(
+    "-d",
+    "--demand-categories",
+    type=click.Choice([x.value for x in DemandCategory]),
+    default=tuple(x.value for x in DemandCategory),
+    show_default=True,
+    multiple=True,
+    callback=lambda *x: tuple(DemandCategory(y) for y in x[2]),
+    help="Demand-based devices to use in time-point selection algorithm",
+)
+@click.option(
+    "-g",
+    "--generation-categories",
+    type=click.Choice([x.value for x in GenerationCategory]),
+    default=tuple(x.value for x in GenerationCategory),
+    show_default=True,
+    multiple=True,
+    callback=lambda *x: tuple(GenerationCategory(y) for y in x[2]),
+    help="Generation-based devices to use in time-point selection algorithm",
+)
+@click.option(
+    "-o",
+    "--output",
+    default="output_time_points",
+    callback=lambda *x: Path(x[2]),
+    help="Output directory",
+)
+@click.option(
+    "--create-new-circuit/--no-create-new-circuit",
+    default=True,
+    is_flag=True,
+    show_default=True,
+    help="Create new circuit with down-selected time points.",
+)
+@click.option(
+    "--check-power-flow/--no-check-power-flow",
+    default=True,
+    is_flag=True,
+    show_default=True,
+    help="Run power flow before and after creating new circuit and check output values.",
+)
+@click.option(
+    "--fix-master-file/--no-fix-master-file",
```

Comment:
Here is the output of the help for this command. Please comment if you would like to change the behavior or text.

```
disco select-time-points --help
Usage: disco select-time-points [OPTIONS] MASTER_FILE

  Select load shape time points in the circuit based on the specified critical
  conditions.

  By default, the Master.dss file is not allowed enable time-series mode.
  Specify --fix-master-file to disable time-series mode and other disallowed
  parameters.
```

Reply:
My take is this:
- The user may or may not understand that we will change settings in their master file to run time-point selection. Maybe they will think that if time-series mode is enabled, we will run in that mode.
- To make sure they understand what happens, make them opt-in to these modifications by passing `--fix-master-file`.

If you all think it's OK to make all of the modifications by default, I can make those changes.

Reply:
Looks good to me.